cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -412,6 +419,81 @@ func (p *BicepProvider) destroyResourceGroups(
 	return nil
 }
 
+func (p *BicepProvider) purgeItems(
+	ctx context.Context,
+	asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress],
+	keyVaults []*azcli.AzCliKeyVault,
+	appConfigs []*azcli.AzCliAppConfig,
+	options DestroyOptions,
+) error {
+	var items string
+	var itemsWarning string
+	if len(appConfigs) > 0 && len(keyVaults) > 0 {
+		items = "Key Vaults and App Configurations"
+		itemsWarning = fmt.Sprintf(""+
+			"\nThis operation will delete %d Key Vaults and %d App Configurations. These %s have soft delete enabled "+
+			"allowing them to be recovered for a period \n"+
+			"of time after deletion. During this period, their names may not be reused.\n"+
+			"You can use argument --purge to skip this confirmation.\n\n",
+			len(keyVaults), len(appConfigs), items)
+	} else if len(keyVaults) > 0 {
+		items = "Key Vaults"
+		itemsWarning = fmt.Sprintf(""+
+			"\nThis operation will delete %d Key Vaults. These %s have soft delete enabled "+
+			"allowing them to be recovered for a period \n"+
+			"of time after deletion. During this period, their names may not be reused.\n"+
+			"You can use argument --purge to skip this confirmation.\n\n",
+			len(keyVaults), items)
+	} else if len(appConfigs) > 0 {
+		items = "App Configurations"
+		itemsWarning = fmt.Sprintf(""+
+			"\nThis operation will delete %d App Configurations. These %s have soft delete enabled "+
+			"allowing them to be recovered for a period \n"+
+			"of time after deletion. During this period, their names may not be reused.\n"+
+			"You can use argument --purge to skip this confirmation.\n\n",
+			len(appConfigs), items)
+	}
+
+	if items != "" && !options.Purge() {
```

Comment:
Consider inverting this if statement and moving to the top of the function to short circuit the function.  When we don't have any items to purge we can quickly exit.  I worry about maintaining this method, etc when we add new resource types in the future.